### PR TITLE
Make clang-tidy a bit happier

### PR DESCRIPTION
**Pull request by mrks**

### CODE REVIEW

File: src/lib/logical_query_plan/abstract_lqp_node.cpp [hunk 1]
```diff
@@ -167,8 +167,8 @@ std::vector<LQPOutputRelation> AbstractLQPNode::output_relations() const {
 
 size_t AbstractLQPNode::output_count() const { return _outputs.size(); }
 
-std::shared_ptr<AbstractLQPNode> AbstractLQPNode::deep_copy(LQPNodeMapping node_mapping) const {
-  return _deep_copy_impl(node_mapping);
+std::shared_ptr<AbstractLQPNode> AbstractLQPNode::deep_copy(LQPNodeMapping extern_node_mapping) const {
```
Comment (mrzzzrm) [inline src/lib/logical_query_plan/abstract_lqp_node.cpp]: "extern"... this is the node mapping of all nodes below this node... so maybe "input_node_mapping"?

File: src/lib/operators/table_scan/base_single_column_table_scan_impl.hpp [hunk 2]
```diff
@@ -25,12 +25,12 @@ class AttributeVectorIterable;
...
 class BaseSingleColumnTableScanImpl : public BaseTableScanImpl, public AbstractColumnVisitor {
  public:
-  BaseSingleColumnTableScanImpl(const std::shared_ptr<const Table>& in_table, const ColumnID left_column_id,
+  BaseSingleColumnTableScanImpl(const std::shared_ptr<const Table>& in_table, const ColumnID column_id,
                                 const PredicateCondition predicate_condition);
 
   std::shared_ptr<PosList> scan_chunk(ChunkID chunk_id) override;
 
-  void handle_column(const ReferenceColumn& left_column, std::shared_ptr<ColumnVisitorContext> base_context) override;
+  void handle_column(const ReferenceColumn& base_column, std::shared_ptr<ColumnVisitorContext> base_context) override;
```
Comment (mrzzzrm) [inline src/lib/operators/table_scan/base_single_column_table_scan_impl.hpp]: it's not a BaseColumn, so this name might be confusing. Just "column" or leave it at "left_column"? (Does clang-tidy actually complain about this?)

File: src/lib/scheduler/topology.hpp [hunk 3]
```diff
@@ -8,7 +8,13 @@
 #include "types.hpp"
 #include "utils/numa_memory_resource.hpp"
 
-namespace boost { namespace container { namespace pmr { class memory_resource; }}}
+namespace boost {
+namespace container {
+namespace pmr {
+class memory_resource;
+}
```
Comment (mrzzzrm) [inline src/lib/scheduler/topology.hpp]: lint doesn't complain about a missing "//namespace pmr"?

File: src/lib/sql/sql_translator.hpp [hunk 4]
```diff
@@ -112,7 +112,7 @@ class SQLTranslator final {
 
   std::shared_ptr<AbstractLQPNode> _translate_create(const hsql::CreateStatement& create_statement);
 
-  std::shared_ptr<AbstractLQPNode> _translate_drop(const hsql::DropStatement& update_statement);
```
Comment (mrzzzrm) [inline src/lib/sql/sql_translator.hpp]: (comment) does clang-tidy actually find this,or was that you?

File: src/lib/statistics/column_statistics.hpp [hunk 5]
```diff
@@ -36,7 +36,7 @@ class ColumnStatistics : public BaseColumnStatistics {
    */
   std::shared_ptr<BaseColumnStatistics> clone() const override;
   FilterByValueEstimate estimate_predicate_with_value(
-      const PredicateCondition predicate_condition, const AllTypeVariant& value,
+      const PredicateCondition predicate_condition, const AllTypeVariant& variant_value,
```
Comment (mrzzzrm) [inline src/lib/statistics/column_statistics.hpp]: We use "value" in other places for instances for AllTypeVariant, I think the name is fine

File: src/test/logical_query_plan/lqp_find_subplan_mismatch_test.cpp [hunk 6]
```diff
@@ -148,4 +148,15 @@ TEST_F(LQPFindSubplanMismatchTest, AdditionalNode) {
   EXPECT_EQ(mismatch->second, additional_predicate_node);
 }
 
+TEST_F(LQPFindSubplanMismatchTest, TypeMismatch) {
+  const auto first_node = PredicateNode::make(between(_query_nodes_rhs.table_a_a, 42, 45));
+  const auto second_node = PredicateNode::make(between(_query_nodes_rhs.table_a_a, 42ll, 45ll));
```
Comment (mrzzzrm) [inline src/test/logical_query_plan/lqp_find_subplan_mismatch_test.cpp]: I think we should use `int64_t{42}` and `int32_t{42}` if we want to make types explicit, imo.

File: src/lib/logical_query_plan/abstract_lqp_node.cpp (hunk 1 above)
Comment (mrks) [reply src/lib/logical_query_plan/abstract_lqp_node.cpp]: Not my name - that already existed in the hpp. I can change it though.

File: src/lib/operators/table_scan/base_single_column_table_scan_impl.hpp (hunk 2 above)
Comment (mrks) [reply src/lib/operators/table_scan/base_single_column_table_scan_impl.hpp]: Mismatching parameter names (see above). Will change.

File: src/lib/scheduler/topology.hpp (hunk 3 above)
Comment (mrks) [reply src/lib/scheduler/topology.hpp]: Surprising, yes.

File: src/lib/sql/sql_translator.hpp (hunk 4 above)
Comment (mrks) [reply src/lib/sql/sql_translator.hpp]: Mismatching parameter name.

File: src/lib/statistics/column_statistics.hpp (hunk 5 above)
Comment (mrks) [reply src/lib/statistics/column_statistics.hpp]: `value` is used in the .cpp once `variant_value` is unpacked. Also mismatching parameter name.

File: src/test/logical_query_plan/lqp_find_subplan_mismatch_test.cpp (hunk 6 above)
Comment (mrks) [reply src/test/logical_query_plan/lqp_find_subplan_mismatch_test.cpp]: Good idea.

**Pull request merged**